src/pip/_internal/operations/install/wheel.py

```diff
@@ -565,19 +567,25 @@ def is_entrypoint_wrapper(name):
         if msg is not None:
             logger.warning(msg)
 
+    @contextlib.contextmanager
+    def _generate_file(path, **kwargs):
+        # type: (str, **Any) -> Iterator[NamedTemporaryFileResult]
+        with adjacent_tmp_file(path, **kwargs) as f:
+            yield f
+        os.chmod(f.name, 0o644)
```

Comment:
I think this should be `0o666 - current_umask()`.

Reply:
TBH this is becoming complex for little benefits. How about

```python
with open(path + ".pip", **kwargs) as f:
    yield f
replace(f.name, path)
```

Reply:
Thanks! I was wondering whether this needs to be more generic, but didn’t know how.

Reply:
@sbidoul Oops, I missed your last comment (stale GitHub tab). Why were we using `adjacent_tmp_file()` in the first place?

Reply:
@uranusjr it came from a [suggestion](https://github.com/pypa/pip/pull/7612#discussion_r373869118) of @chrahunt.
It made sense at first glance but it has generated so much trouble...

Reply:
I traced the `NamedTemporaryFile()` implementation and I think there is a point to use it instead of straight up `open`, but I’m not sure for how much benefit. I’d vote for using the tempfile-chmod method for now (we need to fix this in the 20.1 release), and let @chrahunt chime in before trying to remove the temp file call. At least it would be simple since the calls are now wrapped in a function.